### PR TITLE
Add esbuild-register for typescript extensions

### PR DESCRIPTION
**Pull request by hasparus**
Hi @phated 👋

I felt so smart with my workaround until you mentioned we could do this in a civilized way :P

### CODE REVIEW

**Comment by hasparus** [comment]
image

Test for `.wisp` is failing on my machine. Is this just on my side? Did I break it with my changes somehow?

**Comment by phated** [comment]
Thanks! Wisp is super old and I've always had problems with it. Don't worry about it here

**Comment by hasparus** [comment]
> Thanks! Wisp is super old and I've always had problems with it. Don't worry about it here

All right 👌 LMK if there's something to change in my code then.

File: index.js [hunk 1]
```diff
@@ -146,6 +146,7 @@ var extensions = {
     'typescript-register',
     'typescript-require',
     'sucrase/register/ts',
+    'esbuild-register',
```
[the remark this answers is not in the document]
Comment (hasparus) [reply index.js:149]: Makes sense — so we can pass just `.ts` or `.tsx` to `extensions`.

https://github.com/egoist/esbuild-register/blob/master/src/node.ts#L98

[the remark this answers is not in the document]
Comment (phated) [reply index.js:149]: Yep! We needed to add the "extended" config syntax so we didn't hook extra extensions with babel. We want to be really delicate with the hooking

[the remark this answers is not in the document]
Comment (hasparus) [reply index.js:149]: Ouch, we'll have to require `esbuild-register/dist/node`, right? It's not exposed to the outside.

Something like the following?

```js
    {
      module: 'esbuild-register/dist/node',
      register: function(mod) {
        mod.register({
          extensions: '.ts',
          target: 'node' + process.version.slice(1),
        });
      },
    },
  ```

Or... use `esbuild` directly without `esbuild-register`.

**Comment by hasparus** [comment]
Hey @phated, I just pushed the change with "extended" config syntax.

image

I'm not sure if it's all right, because now the `require-xml` test failed. Could I have broken it or is it a flaky test?

**Review by phated — APPROVED** [review]
Thanks for doing this! 🙇

**Comment by hasparus** [comment]
Hey, I just remembered one thing — I didn't update the README to show this change in [#API](https://github.com/gulpjs/interpret#api) section, and neither did **HRKings** with swc in #74.

**Comment by phated** [comment]
Thanks! I'm actually hoping to finish #36 before we ship v3!

**Pull request merged**